lib/Db/RowMapper.php

```diff
@@ -2,118 +2,312 @@
+
+	/**
+	 * @param array $rowIds
+	 * @param array $columnIds
+	 * @return Row[]
+	 * @throws InternalError
+	 */
+	private function getRows(array $rowIds, array $columnIds): array {
+		$qb = $this->db->getQueryBuilder();
+
+		$qbSqlForColumnTypes = null;
+		foreach ($this->columnsHelper->get(['name']) as $columnType) {
+			$qbTmp = $this->db->getQueryBuilder();
+			$qbTmp->select('*')
+				->from('tables_row_cells_'.$columnType)
+				->where($qb->expr()->in('column_id', $qb->createNamedParameter($columnIds, IQueryBuilder::PARAM_INT_ARRAY, ':columnIds')))
+				->andWhere($qb->expr()->in('row_id', $qb->createNamedParameter($rowIds, IQueryBuilder::PARAM_INT_ARRAY, ':rowsIds')));
+
+			if ($qbSqlForColumnTypes) {
+				$qbSqlForColumnTypes .= ' UNION ALL ' . $qbTmp->getSQL() . ' ';
+			} else {
+				$qbSqlForColumnTypes = '(' . $qbTmp->getSQL();
+			}
+		}
+		$qbSqlForColumnTypes .= ')';
+
+		$qb->select('row_id', 'column_id', 'created_by', 'created_at', 't1.last_edit_by', 't1.last_edit_at', 'value', 'table_id')
+			->from($qb->createFunction($qbSqlForColumnTypes), 't1')
+			->innerJoin('t1', 'tables_row_sleeves', 'rowSleeve', 'rowSleeve.id = t1.row_id');
+
 		try {
-			$columnQbClassName = 'OCA\Tables\Db\ColumnTypes\\';
-			$type = explode("-", $filter['columnType'])[0];
+			$result = $this->db->executeQuery($qb->getSQL(), $qb->getParameters(), $qb->getParameterTypes());
+		} catch (Exception $e) {
+			$this->logger->error($e->getMessage(), ['exception' => $e]);
+			throw new InternalError(get_class($this) . ' - ' . __FUNCTION__ . ': ' . $e->getMessage(), );
+		}
 
-			$columnQbClassName .= ucfirst($type).'ColumnQB';
+		try {
+			$sleeves = $this->rowSleeveMapper->findMultiple($rowIds);
+		} catch (Exception $e) {
+			$this->logger->error($e->getMessage(), ['exception' => $e]);
+			throw new InternalError(get_class($this) . ' - ' . __FUNCTION__ . ': '.$e->getMessage());
+		}
 
-			/** @var IColumnTypeQB $columnQb */
-			$columnQb = Server::get($columnQbClassName);
-			return $columnQb->addWhereFilterExpression($qb, $filter, $filterId);
-		} catch (NotFoundExceptionInterface|ContainerExceptionInterface $e) {
-			$this->logger->debug('Column type query builder class not found');
+		return $this->parseEntities($result, $sleeves);
+	}
+
+	/**
+	 * @throws InternalError
+	 */
+	private function addFilterToQuery(IQueryBuilder &$qb, array $filters, string $userId): void {
+		// TODO move this into service
+		$this->replaceMagicValues($filters, $userId);
+
+		if (count($filters) > 0) {
+			$qb->andWhere(
+				$qb->expr()->orX(
+					...$this->getFilterGroups($qb, $filters)
+				)
+			);
 		}
-		return null;
 	}
 
-	private function getInnerFilterExpressions($qb, $filterGroup, int $groupIndex): array {
-		$innerFilterExpressions = [];
-		foreach ($filterGroup as $index => $filter) {
-			$innerFilterExpressions[] = $this->buildFilterByColumnType($qb, $filter, $groupIndex.$index);
+	private function replaceMagicValues(array &$filters, string $userId): void {
```

Comment:
reading `Magic` in code is scary. I would not guess what i means and does. From the resolveSearchValue line my guess would have been it is looking up a user. Then i looked into  that function… What do you think about `replacePlaceholderValues()`?